docs/src/main/asciidoc/_admin-guide/management-api/Description_of_the_Management_Model.adoc

```diff
@@ -516,3 +522,27 @@ operation will go ahead and restart the runtime service. Otherwise
 executing the operation will put the server into a `reload-required`
 state. (See the discussion of `all-services` above for more on the
 `reload-required` state.)
+
+[[Expression_Resolution]]
+== Expression Resolution
+When resolving an expression in the model the following locations are checked.
+For this example we will use the expression `${my.example-expr}`.
+
+* First we check if there is a system property with the name
+`my.example-expr`.
+If there is, we use its value as the result of the resolution.
+If not, we continue checking the next locations.
+* We convert the name `my.example-expr` to upper case, and replace all
+non-alphanumeric characters with underscores, ending up with
+`MY_EXAMPLE_EXPR`. We check if there is an environment variable with that
+name. If there is, we use its value as the result of the resolution.
+If not, we continue checking the next location.
+* We append `env.` onto the original name, i,e, `env.my.example-expr`
```

Comment:
Does this mean checking for literally `env.my.example-expr` environment variable, or is the SCREAMING_SNAKE_CASE conversion applied and we check for `ENV_MY_EXAMPLE_EXPR`

Reply:
Actually this is slightly wrong, and was what always happened.

What we do is check if the original name starts with 'env.', and if it does we drop the suffix, and use the remainder as the env var that we look up.  So `env.blah` becomes `blah` and `env.HI_THERE` becomes `HI_THERE`

Reply:
So maybe a clarification is in order, so that it is in line with analysis [1] 
https://github.com/wildfly/wildfly-proposals/blob/3f1ec1e5ae9aadc2cd0b7dc34c4e7fe4327e3086/management/WFCORE-5464_Check_Env_Vars_On_Expression_Resolution.adoc#hard-requirements

Reply:
I pushed a fix

Reply:
Much clearer now, thank you